wire/internal/wire/wire_test.go

```diff
@@ -59,116 +61,108 @@ func TestWire(t *testing.T) {
 	}
 	wd := filepath.Join(magicGOPATH(), "src")
 
-	t.Run("Generate", func(t *testing.T) {
-		if _, err := os.Stat(filepath.Join(build.Default.GOROOT, "bin", "go")); err != nil {
-			t.Skip("go toolchain not available:", err)
-		}
-		for _, test := range tests {
-			test := test
-			t.Run(test.name, func(t *testing.T) {
-				t.Parallel()
-
-				// Run Wire from a fake build context.
-				bctx := test.buildContext()
-				gen, errs := Generate(bctx, wd, test.pkg)
-				if len(gen) > 0 {
-					defer t.Logf("wire_gen.go:\n%s", gen)
+	if _, err := os.Stat(filepath.Join(build.Default.GOROOT, "bin", "go")); err != nil {
+		t.Skip("go toolchain not available:", err)
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			// Run Wire from a fake build context.
+			bctx := test.buildContext()
+			gen, errs := Generate(bctx, wd, test.pkg)
+			if len(gen) > 0 {
+				defer t.Logf("wire_gen.go:\n%s", gen)
+			}
+			if len(errs) > 0 {
+				for _, e := range errs {
+					t.Log(e)
 				}
-				if len(errs) > 0 {
-					for _, e := range errs {
-						t.Log(e)
-					}
-					if !test.wantError {
-						t.Fatal("Did not expect errors.")
-					}
-					for _, s := range test.wantErrorStrings {
-						if !errorListContains(errs, s) {
-							t.Errorf("Errors did not contain %q", s)
-						}
-					}
-					return
+				if !test.wantError {
+					t.Fatal("Did not expect errors.")
 				}
-				if len(errs) == 0 && test.wantError {
-					t.Fatal("wirego succeeded; want error")
+				for _, s := range test.wantErrorStrings {
+					if !errorListContains(errs, s) {
+						t.Errorf("Errors did not contain %q", s)
+					}
 				}
+				return
+			}
+			if len(errs) == 0 && test.wantError {
+				t.Fatal("wirego succeeded; want error")
+			}
 
-				// Find the absolute import path, since test.pkg may be a relative
-				// import path.
-				genPkg, err := bctx.Import(test.pkg, wd, build.FindOnly)
-				if err != nil {
-					t.Fatal(err)
+			goldenDir := filepath.Join("testdata", test.name, ".golden")
+			if err := os.MkdirAll(goldenDir, 0777); err != nil {
+				t.Fatalf("Failed to make golden dir: %v", err)
+			}
+			goldenFile := filepath.Join(goldenDir, "wire_gen.go")
+			if *setup.Record {
+				// Record ==> Check wire output with go build, and save golden file if it looks good.
+				goBuildCheck(t, test, wd, bctx, gen)
+				if err := ioutil.WriteFile(goldenFile, gen, 0666); err != nil {
+					t.Fatalf("Failed to write golden file: %v", err)
 				}
-
-				// Run a `go build` with the generated output.
-				gopath, err := ioutil.TempDir("", "wire_test")
+			} else {
+				// Replay ==> Load golden file and compare to generated result.
```

Comment:
Since determinism was a concern in the previous version, maybe add a comment here noting that if Generate isn't deterministic, the replay test will become flaky and catch that (although the --record run won't).

Reply:
Done.